src/TaskQueue.js

```diff
@@ -0,0 +1,44 @@
+const iterate = require('taskcluster-lib-iterate');
+const assert = require('assert');
+class TaskQueue {
+  constructor(cfg, queue) {
+    assert(cfg.worker.workerId, 'Worker ID is required');
+    assert(cfg.worker.workerType, 'Worker type is required');
+    assert(cfg.worker.workerGroup, 'Worker group is required');
+    assert(cfg.worker.provisionerId, 'Provisioner ID is required');
+    assert(queue, 'Instance of taskcluster queue is required');
+    this.queue = queue;
+    this.workerType = cfg.worker.workerType;
+    this.provisionerId = cfg.worker.provisionerId;
+    this.workerGroup = cfg.worker.workerGroup;
+    this.workerId = cfg.worker.workerId;
+  }
+  async startWorker() {
+
+    while (true) {
+      await this.claimTasks();
+    }
+  }
+
+  async claimTasks() {
+    let result = await this.queue.claimWork(this.provisionerId, this.workerType, {
+      tasks: 1,
+      workerGroup: this.workerGroup,
+      workerId: this.workerId,
+    });
+    let stat = '';
+    if (Object.keys(result.tasks[0].task.payload).length===0) {
```

Comment:
Note that `result.tasks` might be empty.  It'll be best to check this and then get the task if one exists (otherwise just return):
```javascript
  if (result.tasks.length === 0) {
    return;
  }
  const task = result.tasks[0];
  if (Object.keys(task.payload).length === 0) { ...
```